### PR TITLE
Add missing interface to set init processes Umask

### PR DESCRIPTION
**Pull request by rhatdan**
Umask is a field specified in the runtime spec, but we don't
have a method to set it in runtime-tools.  Some users might
want to modify the default Umask of a container.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mrunalp PTAL

**Comment by rhatdan** [comment]
@q384566678 @wking @cyphar PTAL

**Comment by rhatdan** [comment]
@haircommander PTAL

**Comment by q384566678** [comment]
I think we need to add corresponding options in the generate function to call this function.

**Comment by rhatdan** [comment]
@q384566678 Added.

**Comment by haircommander** [comment]
LGTM, thanks @rhatdan

**Comment by q384566678** [comment]
LGTM

**Comment by rhatdan** [comment]
@caniszczyk @crosbymichael @dqminh @hqhq @liangchenye @mashimiao @mrunalp @philips @tianon @vbatts @vishh PTAL

**Comment by philips** [comment]
I am no longer a maintainer FYI 👍

**Comment by rhatdan** [comment]
@philips Your name is still listed, sorry for bothering you.

**Comment by mrunalp** [comment]
LGTM

**Comment by philips** [comment]
@rhatdan where is it listed? I think i am gone from maintainers.

**Comment by rhatdan** [comment]
@philips https://pullapprove.com/opencontainers/runtime-tools/pull-request/704/reviews/#group:runtime-tools

**Comment by cyphar** [comment]
Yeah the @opencontainers/runtime-tools-maintainers list needs to be updated. /cc @caniszczyk?

File: generate/generate.go [hunk 1]
```diff
@@ -444,6 +444,12 @@ func (g *Generator) SetProcessUsername(username string) {
 	g.Config.Process.User.Username = username
 }
 
+// SetProcessUmask sets g.Config.Process.User.Umask.
+func (g *Generator) SetProcessUmask(umask uint32) {
+	g.initConfigProcess()
+	g.Config.Process.User.Umask = umask
+}
```
Comment (saschagrunert) [inline generate/generate.go:451]: This has changed to be an `*uint32` with https://github.com/opencontainers/runtime-spec/commit/d3f079a3fd5c503631012f71bc0fda66dcc9e755, which breaks compatibility between the tools and the spec of both latest master branches.

Comment (kolyshkin) [reply generate/generate.go:451]: Just found the same thing :)

Comment (kolyshkin) [reply generate/generate.go:451]: So, there is a PR to fix it (https://github.com/opencontainers/runtime-tools/pull/708) but apparently it requires a new release of runtime-specs, for which there is an issue opened (https://github.com/opencontainers/runtime-spec/issues/1052) but it's not yet done so...

**Pull request merged**